lib/bundler/definition.rb

```diff
@@ -533,7 +533,14 @@ def converge_sources
 
     def converge_dependencies
       (@dependencies + @locked_deps).each do |dep|
-        dep.source = sources.get(dep.source) if dep.source
+        locked_source = @locked_deps.select {|d| d.name == dep.name }.last
```

Comment:
It there a better way to find if the only difference of two dependencies is the sources' path? I've used dependency's name to find (which I assume is the identifier)